### PR TITLE
Add note to SSC+Server about garbage collection

### PR DESCRIPTION
**Pull request by ngrayluna**
## Description

What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Comment by ngrayluna** [comment]
Preview:
Screen Shot 2023-11-06 at 8 45 25 AM

File: docs/guides/hosting/secure-storage-connector.md [hunk 1]
```diff
@@ -19,3 +19,7 @@ An error or warning will appear at the bottom of the page if there are issues ac
 ![](/images/hosting/prod_setup_secure_storage.png)
 
 Only system administrators have the permissions to configure the secure storage connector. The same cloud storage bucket can be used amongst multiple teams by selecting an existing cloud storage bucket from the dropdown.
+
+:::note
+W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for W&B Server, see the [Delete artifacts](../artifacts/delete-artifacts.md) page.
+:::
```
Comment (noaleetz) [inline docs/guides/hosting/secure-storage-connector.md:25]: Hey @ngrayluna - we need to add a section in the actual GC/deletion docs explaining how to enable GC for SSC (BYOB) - this different than for W&B Server. For dedicated cloud w/ SSC/BYOB - versioning needs to be enabled (no env variable needs to be set). 

Can we add this into this PR? This is re the comment I left [here](https://github.com/wandb/docodile/pull/462#issuecomment-1789846969). Then, the section above needs to be edited to link to that section (enabling GC for dedicated cloud on BYOB). 

TLDR 
- cloud and dedicated cloud (without BYOB) - GC is enabled by default
- dedicated cloud WITH BYOB - no env variable to set, **only** enabling versioning
- server - set env var AND enable versioning (both required)
we need to distinguish the requirements for dedicated w BYOB and Server in the GC docs and then link to relevant section

**Review by noaleetz — CHANGES_REQUESTED** [review]
left a comment re distinguishing requirements for GC for byob and server

File: docs/guides/artifacts/delete-artifacts.md [hunk 2]
```diff
@@ -16,12 +16,12 @@ Delete artifacts interactively with the App UI or programmatically with the W&B
 The contents of the artifact remain as a soft-delete, or pending deletion state, until a regularly run garbage collection process reviews all artifacts marked for deletion. The garbage collection process deletes associated files from storage if the artifact and its associated files are not used by a previous or subsequent artifact versions. 
 
 
-:::note
+<!-- :::note
```
Comment (ngrayluna) [inline docs/guides/artifacts/delete-artifacts.md]: Note: This is commented out. Does not show up on docs.

**Comment by ngrayluna** [comment]
Preview of "Delete artifact" page w/ GC section: 
[Delete W&B Artifacts.pdf](https://github.com/wandb/docodile/files/13273556/Delete.W.B.Artifacts.pdf)

File: docs/guides/hosting/secure-storage-connector.md [hunk 3]
```diff
@@ -19,3 +19,7 @@ An error or warning will appear at the bottom of the page if there are issues ac
 ![](/images/hosting/prod_setup_secure_storage.png)
 
 Only system administrators have the permissions to configure the secure storage connector. The same cloud storage bucket can be used amongst multiple teams by selecting an existing cloud storage bucket from the dropdown.
+
+:::note
+W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for secure storage container see the [How to enable garbage collection based on how you host W&B](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-you-host-wb) page.
```
Comment (noaleetz) [inline docs/guides/hosting/secure-storage-connector.md]: we should specify that this is how to hard-deletion (not delete in general via. delete() in API or delete in UI) since deletion guides focuses on soft delete. This

File: docs/guides/artifacts/delete-artifacts.md [hunk 4]
```diff
@@ -118,3 +118,24 @@ artifact_name = "<>"  # provide artifact name
...
+* Enable bucket versioning if you use [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html), [GCP](https://cloud.google.com/storage/docs/object-versioning) or any other storage provider such as [Minio](https://min.io/docs/minio/linux/administration/object-management/object-versioning.html#enable-bucket-versioning). If you use Azure, [enable soft deletion](https://learn.microsoft.com/en-us/azure/storage/blobs/soft-delete-blob-overview).
+  :::note
+  Soft deletion in Azure is equivalent to bucket versioning in other storage providers.
+  :::
+
+The following table describes how to satisfy requirements to enable garbage collection based on your deployment type. The `X` indicates you must satisfy the requirement.
+
+|                                                | Environment variable | Enable versioning | 
+| -----------------------------------------------| ---------------------| ----------------- |
+| Dedicated cloud with secure storage container  |                      | X                 |
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md]: For dedicated, the customer should ask their account team to enable the environment variable, and then configure the bucket versioning / soft deletion in their bucket (only for AWS/Azure). We should reflect that, so customer doesn't think that they could somehow edit that env variable themselves.

File: docs/guides/artifacts/delete-artifacts.md [hunk 5]
```diff
@@ -118,3 +118,24 @@ artifact_name = "<>"  # provide artifact name
...
 artifact.collection.delete()
 ```
+
+
+## How to enable garbage collection based on how you host W&B 
+
+You might need to take additional steps to enable garbage collection if you use W&B Server. Based on your deployment type, you might need to satisfy the following requirements:
+
+* Set the `GORILLA_ARTIFACT_GC_ENABLED` environment variable to true: `GORILLA_ARTIFACT_GC_ENABLED=true`
+* Enable bucket versioning if you use [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html), [GCP](https://cloud.google.com/storage/docs/object-versioning) or any other storage provider such as [Minio](https://min.io/docs/minio/linux/administration/object-management/object-versioning.html#enable-bucket-versioning). If you use Azure, [enable soft deletion](https://learn.microsoft.com/en-us/azure/storage/blobs/soft-delete-blob-overview).
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md:124]: Should probably indicate that there's no secure storage connector with Azure, so it doesn't apply to dedicated cloud. Maybe as a note.

File: docs/guides/hosting/secure-storage-connector.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ The team-level secure storage connector allows teams within W&B to utilize a sep
 This feature is currently only available for Google Cloud Platform and Amazon Web Services. To request a license to enable this feature, email contact@wandb.com.
 :::
 
-A cloud storage bucket can be configured only once for a team at the time of team creation. To provision a bucket, we recommend using our terraform for [AWS](https://github.com/wandb/terraform-aws-wandb/tree/main/modules/secure_storage_connector) or [GCP](https://github.com/wandb/terraform-google-wandb/tree/main/modules/secure_storage_connector).
+A cloud storage bucket can be configured only once for a team at the time of team creation. To provision a bucket, W&B recommends that you use a Terraform configuration managed by W&B for [AWS](https://github.com/wandb/terraform-aws-wandb/tree/main/modules/secure_storage_connector) or [GCP](https://github.com/wandb/terraform-google-wandb/tree/main/modules/secure_storage_connector).
```
Comment (abhinavg6) [inline docs/guides/hosting/secure-storage-connector.md]:
```suggestion
A cloud storage bucket can be configured only once for a team at the time of team creation. To provision a bucket, W&B recommends that you use a Terraform module managed by W&B for [AWS](https://github.com/wandb/terraform-aws-wandb/tree/main/modules/secure_storage_connector) or [GCP](https://github.com/wandb/terraform-google-wandb/tree/main/modules/secure_storage_connector).
```

File: docs/guides/hosting/secure-storage-connector.md (hunk 1 above)
Comment (abhinavg6) [reply docs/guides/hosting/secure-storage-connector.md:25]: I added one comment related to that table in the other file.

File: docs/guides/artifacts/delete-artifacts.md [hunk 7]
```diff
@@ -118,3 +118,24 @@ artifact_name = "<>"  # provide artifact name
 artifact = api.artifact(artifact_name)
 artifact.collection.delete()
 ```
+
+
+## How to enable garbage collection based on how you host W&B
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md]:
```suggestion
## How to enable garbage collection based on how W&B is hosted
```

File: docs/guides/hosting/secure-storage-connector.md [hunk 8]
```diff
@@ -10,7 +10,7 @@ The team-level secure storage connector allows teams within W&B to utilize a sep
...
 
 Select **External Storage** when you create a team tp configure a cloud storage bucket. Select your provider and fill out your bucket name and storage encryption key ID, if applicable, and select **Create Team**.
```
Comment (abhinavg6) [inline docs/guides/hosting/secure-storage-connector.md]:
```suggestion
Select **External Storage** when you create a team to configure a cloud storage bucket. Select your provider and fill out your bucket name and storage encryption key ID, if applicable, and select **Create Team**.
```

File: docs/guides/artifacts/delete-artifacts.md (hunk 5 above)
Comment (abhinavg6) [reply docs/guides/artifacts/delete-artifacts.md:124]: Basically it would apply for Azure only with Customer managed cloud and for the deployment-level bucket (there are no team-level Azure buckets yet).

**Review by abhinavg6 — CHANGES_REQUESTED** [review]
Added some comments and requested a few edits.

File: docs/guides/artifacts/delete-artifacts.md (hunk 4 above)
Comment (noaleetz) [reply docs/guides/artifacts/delete-artifacts.md]: @szymon-piechowicz-wandb @jlzhao27 I had understood this isn't something the user needs to do and that we by default turn on this env variable for dedicated, no? If not, I think we should enable it by default. It is going to introduce a lot of friction if they need to go to their account team and the account team has to figure out how to do this, etc. Is there a reason not to enable it by default for all dedicated cloud instances?

File: docs/guides/artifacts/delete-artifacts.md (hunk 5 above)
Comment (noaleetz) [reply docs/guides/artifacts/delete-artifacts.md:124]: Gotcha - good point - @ngrayluna should we add a note in the table or at the end that if you're on Azure the row dedicated cloud with SSC is not relevant for you - dedicated cloud with SSC is not an option for Azure buckets

Comment (ngrayluna) [reply docs/guides/artifacts/delete-artifacts.md:124]: We have a note at the top of the secure storage connector page: https://docs.wandb.ai/guides/hosting/secure-storage-connector Does that suffice? 
Screen Shot 2023-11-08 at 11 17 27 AM

File: docs/guides/artifacts/delete-artifacts.md (hunk 4 above)
Comment (ngrayluna) [reply docs/guides/artifacts/delete-artifacts.md]: Added: 

"By default, dedicated cloud with secure storage connector can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable. 

In addition, W&B encourages dedicated cloud users (with secure storage connector) to ask their account team to enable the environment variable before you enable bucket versioning (AWS, GCP, and so on) or soft deletion (Azure).
"

File: docs/guides/artifacts/delete-artifacts.md (hunk 5 above)
Comment (abhinavg6) [reply docs/guides/artifacts/delete-artifacts.md:124]: Yeah I had seen that, but I feel we should add a simple note here as well since we're talking about enabling stuff on that cloud. Just so that if anyone just reads this page and doesn't go to the W&B hosting page, they can know right here itself.

File: docs/guides/artifacts/delete-artifacts.md [hunk 9]
```diff
@@ -118,3 +110,33 @@ artifact_name = "<>"  # provide artifact name
...
+
+|                                                | Environment variable | Enable versioning | 
+| -----------------------------------------------| ---------------------| ----------------- |
+| Dedicated cloud with secure storage connector  |                      | X                 |
+| Customer-managed cloud                         | X                    | X                 |
+| Customer managed on-prem                       | X                    | X                 |
+ 
+
+
+By default, dedicated cloud with secure storage connector can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable.
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md]:
```suggestion
By default,  `GORILLA_ARTIFACT_GC_ENABLED` environment variable is not enabled on dedicated cloud with secure storage connector. Ask your account team to enable the environment variable, preferably before you enable versioning for your bucket. 
```

File: docs/guides/artifacts/delete-artifacts.md [hunk 10]
```diff
@@ -118,3 +110,33 @@ artifact_name = "<>"  # provide artifact name
...
+| -----------------------------------------------| ---------------------| ----------------- |
+| Dedicated cloud with secure storage connector  |                      | X                 |
+| Customer-managed cloud                         | X                    | X                 |
+| Customer managed on-prem                       | X                    | X                 |
+ 
+
+
+By default, dedicated cloud with secure storage connector can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable. 
+
+In addition, W&B encourages dedicated cloud users (with secure storage connector) to ask their account team to enable the environment variable before you enable bucket versioning (AWS, GCP, and so on) or soft deletion (Azure).
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md]: I think we could remove this, if you accept the above suggestion,

File: docs/guides/artifacts/delete-artifacts.md [hunk 11]
```diff
@@ -118,3 +110,33 @@ artifact_name = "<>"  # provide artifact name
...
+  Soft deletion in Azure is equivalent to bucket versioning in other storage providers.
+  :::
+
+The following table describes how to satisfy requirements to enable garbage collection based on your deployment type. 
+
+The `X` indicates you must satisfy the requirement:
+
+|                                                | Environment variable | Enable versioning | 
+| -----------------------------------------------| ---------------------| ----------------- |
+| Dedicated cloud with secure storage connector  |                      | X                 |
```
Comment (abhinavg6) [inline docs/guides/artifacts/delete-artifacts.md]: Actually enable versioning should be X for dedicated cloud with secure storage connector too, right?

File: docs/guides/hosting/secure-storage-connector.md [hunk 12]
```diff
@@ -10,12 +10,16 @@ The team-level secure storage connector allows teams within W&B to utilize a sep
...
+Select **External Storage** when you create a team to configure a cloud storage bucket. Select your provider and fill out your bucket name and storage encryption key ID, if applicable, and select **Create Team**.
 
 An error or warning will appear at the bottom of the page if there are issues accessing the bucket or the bucket has invalid settings.
 
 ![](/images/hosting/prod_setup_secure_storage.png)
 
 Only system administrators have the permissions to configure the secure storage connector. The same cloud storage bucket can be used amongst multiple teams by selecting an existing cloud storage bucket from the dropdown.
+
+:::note
+W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for secure storage container see the [How to enable garbage collection based on how you host W&B](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-you-host-wb) page.
```
Comment (abhinavg6) [inline docs/guides/hosting/secure-storage-connector.md]:
```suggestion
W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for secure storage container see the [How to enable garbage collection based on how you host W&B](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-wb-is-hosted) page.
```

**Review by abhinavg6 — CHANGES_REQUESTED** [review]
Sorry, added a few more comments requesting changes.

File: docs/guides/artifacts/delete-artifacts.md (hunk 11 above)
Comment (ngrayluna) [reply docs/guides/artifacts/delete-artifacts.md]: It is:
Screen Shot 2023-11-08 at 1 42 28 PM

File: docs/guides/artifacts/delete-artifacts.md [hunk 13]
```diff
@@ -126,17 +126,22 @@ The following table describes how to satisfy requirements to enable garbage coll
...
+| Shared cloud with secure storage connector     |                         | X                 | 
+| Dedicated cloud                                |                         |                   | 
+| Dedicated cloud with secure storage connector  |                         | X                 | 
+| Customer-managed cloud                         | X                       | X                 | 
+| Customer managed on-prem                       | X                       | X                 |
  
 
 
-By default, dedicated cloud with secure storage connector can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable. 
+By default, dedicated cloud with [secure storage connector](../hosting/secure-storage-connector.md) can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable, preferably before you enable versioning for your bucket.
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md]: this can be removed. we will be turning it on for those we can set the var for (we be on by default after terraform changes go through)

File: docs/guides/artifacts/delete-artifacts.md [hunk 14]
```diff
@@ -126,17 +126,22 @@ The following table describes how to satisfy requirements to enable garbage coll
...
-| Customer-managed cloud                         | X                    | X                 |
-| Customer managed on-prem                       | X                    | X                 |
+|                                                | Environment variable    | Enable versioning | 
+| -----------------------------------------------| ------------------------| ----------------- | 
+| Shared cloud                                   |                         |                   | 
+| Shared cloud with secure storage connector     |                         | X                 | 
+| Dedicated cloud                                |                         |                   | 
+| Dedicated cloud with secure storage connector  |                         | X                 | 
+| Customer-managed cloud                         | X                       | X                 | 
+| Customer managed on-prem                       | X                       | X                 |
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md:140]: I am not sure if this is the best place for this table - this needs to go in the same place we show people the instructions for enabling versioning and setting the environment variable. This should likely be in the section where we talk about enabling GC on different hosting types.

File: docs/guides/hosting/secure-storage-connector.md [hunk 15]
```diff
@@ -21,5 +21,5 @@ An error or warning will appear at the bottom of the page if there are issues ac
 Only system administrators have the permissions to configure the secure storage connector. The same cloud storage bucket can be used amongst multiple teams by selecting an existing cloud storage bucket from the dropdown.
 
 :::note
-W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for secure storage container see the [How to enable garbage collection based on how you host W&B](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-you-host-wb) page.
+W&B uses a garbage collection process to delete W&B artifacts. For more information on how W&B deletes artifacts, and for information on how to enable garbage collection for secure storage container see the [How to enable garbage collection based on how you host W&B](../artifacts/delete-artifacts.md#how-to-enable-garbage-collection-based-on-how-wb-is-hosted) page.
```
Comment (noaleetz) [inline docs/guides/hosting/secure-storage-connector.md]: for secure storage container OR on customer-managed deployment types

File: docs/guides/artifacts/delete-artifacts.md (hunk 14 above)
Comment (noaleetz) [reply docs/guides/artifacts/delete-artifacts.md:140]: We should note in these docs - if you are on saas/shared cloud or dedicated cloud - garbage collection is enabled by default (you do not have to do anything). Unless you use SSC, in that case, learn more in this section (how to enable gc based on how you host gc section, containing the matrix)

File: docs/guides/artifacts/delete-artifacts.md [hunk 16]
```diff
@@ -126,17 +126,22 @@ The following table describes how to satisfy requirements to enable garbage coll
...
 
 
-By default, dedicated cloud with secure storage connector can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable. 
+By default, dedicated cloud with [secure storage connector](../hosting/secure-storage-connector.md) can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable. Ask your account team to enable the environment variable, preferably before you enable versioning for your bucket.
 
-In addition, W&B encourages dedicated cloud users (with secure storage connector) to ask their account team to enable the environment variable before you enable bucket versioning (AWS, GCP, and so on) or soft deletion (Azure).
 
+:::note
+Secure storage connector is currently only available for Google Cloud Platform and Amazon Web Services.
+:::
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md:146]: since we are removing any note of ssc in this guide - this should be moved into where we mention enabling GC for SSC

File: docs/guides/hosting/secure-storage-connector.md (hunk 15 above)
Comment (noaleetz) [reply docs/guides/hosting/secure-storage-connector.md]: the doc linked should have the table which covers both SSC and customer-managed / local deployments

**Review by noaleetz — CHANGES_REQUESTED** [review]
left comments

File: docs/guides/artifacts/delete-artifacts.md (hunk 14 above)
Comment (ngrayluna) [reply docs/guides/artifacts/delete-artifacts.md:140]: I agree that this section, nor this table,  should not exist here. It should go in a separate page in Server. E.g. [See this thread](https://weightsandbiases.slack.com/archives/C064WHTC6L9/p1699316278423919).

My thinking is that we leave it here for now and make a note to remove it .

File: docs/guides/hosting/secure-storage-connector.md (hunk 15 above)
Comment (ngrayluna) [reply docs/guides/hosting/secure-storage-connector.md]: Can't link to a table directly, unfortunately. Only section titles

File: docs/guides/artifacts/delete-artifacts.md (hunk 16 above)
Comment (ngrayluna) [reply docs/guides/artifacts/delete-artifacts.md:146]: This is a to do for the future. Right now I am not creating another page for this PR

**Comment by ngrayluna** [comment]
Updated:
[Delete W&B Artifacts.pdf](https://github.com/wandb/docodile/files/13302089/Delete.W.B.Artifacts.pdf)

Note: we are leaving how to enable GC in artifacts docs for now. This will be removed to Server in the future.

File: docs/guides/artifacts/delete-artifacts.md [hunk 17]
```diff
@@ -15,14 +15,6 @@ Delete artifacts interactively with the App UI or programmatically with the W&B
...
 
-
-:::note
-Garbage collection is not automatically enabled for W&B Server. Satisfy the following requirements to enable garbage collection in W&B Server:
-* Set the `GORILLA_ARTIFACT_GC_ENABLED` environment variable to true: `GORILLA_ARTIFACT_GC_ENABLED=true`
-* Enable bucket versioning if you use [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html) or [GCP](https://cloud.google.com/storage/docs/object-versioning).
-* [Enable soft delete for blobs if you use Azure](https://learn.microsoft.com/en-us/azure/storage/blobs/soft-delete-blob-overview).
-:::
-
 The sections in this page describe how to delete specific artifact versions, how to delete an artifact collection, how to delete artifacts with and without aliases, and more. You can schedule when artifacts are deleted from W&B with TTL policies. For more information, see [Manage data retention with Artifact TTL policy](./ttl.md).
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md:18]:
```suggestion
The sections in this page describe how to delete specific artifact versions, how to delete an artifact collection, how to delete artifacts with and without aliases, and more. You can schedule when artifacts are deleted from W&B with TTL policies. For more information, see [Manage data retention with Artifact TTL policy](./ttl.md). Note that whether you schedule an artifact for deletion with a TTL policy, or delete it immediately via the SDK or UI, both result in a soft-deletion and be subject to garbage collection to be transitioned to a hard-deleted state. 
```

File: docs/guides/artifacts/delete-artifacts.md [hunk 18]
```diff
@@ -118,3 +110,38 @@ artifact_name = "<>"  # provide artifact name
...
+  :::
+
+The following table describes how to satisfy requirements to enable garbage collection based on your deployment type. 
+
+The `X` indicates you must satisfy the requirement:
+
+|                                                | Environment variable    | Enable versioning | 
+| -----------------------------------------------| ------------------------| ----------------- | 
+| Shared cloud                                   |                         |                   | 
+| Shared cloud with secure storage connector     |                         | X                 |
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md]:
```suggestion
| Shared cloud with [secure storage connector](https://docs.wandb.ai/guides/hosting/secure-storage-connector)     |                         | X                 | 
```

File: docs/guides/artifacts/delete-artifacts.md [hunk 19]
```diff
@@ -118,3 +110,38 @@ artifact_name = "<>"  # provide artifact name
...
+The following table describes how to satisfy requirements to enable garbage collection based on your deployment type. 
+
+The `X` indicates you must satisfy the requirement:
+
+|                                                | Environment variable    | Enable versioning | 
+| -----------------------------------------------| ------------------------| ----------------- | 
+| Shared cloud                                   |                         |                   | 
+| Shared cloud with secure storage connector     |                         | X                 | 
+| Dedicated cloud                                |                         |                   | 
+| Dedicated cloud with secure storage connector  |                         | X                 |
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md]:
```suggestion
| Dedicated cloud with [secure storage connector](https://docs.wandb.ai/guides/hosting/secure-storage-connector)  |                         | X                 | 
```

File: docs/guides/artifacts/delete-artifacts.md [hunk 20]
```diff
@@ -118,3 +110,38 @@ artifact_name = "<>"  # provide artifact name
...
+| Shared cloud                                   |                         |                   | 
+| Shared cloud with secure storage connector     |                         | X                 | 
+| Dedicated cloud                                |                         |                   | 
+| Dedicated cloud with secure storage connector  |                         | X                 | 
+| Customer-managed cloud                         | X                       | X                 | 
+| Customer managed on-prem                       | X                       | X                 |
+ 
+
+
+By default, dedicated cloud with [secure storage connector](../hosting/secure-storage-connector.md) can not edit the `GORILLA_ARTIFACT_GC_ENABLED` environment variable.
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md]: I am wondering if we still need this note? Is the point here to emphasize that we need to enable it for you if on dedicated cloud?

File: docs/guides/artifacts/delete-artifacts.md (hunk 5 above)
Comment (noaleetz) [reply docs/guides/artifacts/delete-artifacts.md:124]: This is a nit but I don't think we need to note that SSC is not available for azure. If the user is already on SSC they will already know this, and if they're not they will definitely go to the SSC page to learn more about it and figure that out. this page is on deleting artifacts so it is a bit redundant to add a warning on SSC availability in here. 

But if folks feel strongly I will concede on this one :). It is a nit. cc @abhinavg6 @ngrayluna

File: docs/guides/artifacts/delete-artifacts.md (hunk 20 above)
Comment (noaleetz) [reply docs/guides/artifacts/delete-artifacts.md]: I am thinking this might not be necessary to add since for both dedicated rows we say that setting the env var is not a requirement (since we will handle this for dedicated instances). So a user would not attempt to try to set it themselves.

**Review by noaleetz — CHANGES_REQUESTED** [review]
left a few more suggestions including adding hyperlinks for SSC in table

**Comment by ngrayluna** [comment]
Updated: 
[Delete W&B Artifacts.pdf](https://github.com/wandb/docodile/files/13310888/Delete.W.B.Artifacts.pdf)

File: docs/guides/artifacts/delete-artifacts.md [hunk 21]
```diff
@@ -15,16 +15,12 @@ Delete artifacts interactively with the App UI or programmatically with the W&B
...
 
+The sections in this page describe how to delete specific artifact versions, how to delete an artifact collection, how to delete artifacts with and without aliases, and more. You can schedule when artifacts are deleted from W&B with TTL policies. For more information, see [Manage data retention with Artifact TTL policy](./ttl.md).
 
 :::note
-Garbage collection is not automatically enabled for W&B Server. Satisfy the following requirements to enable garbage collection in W&B Server:
-* Set the `GORILLA_ARTIFACT_GC_ENABLED` environment variable to true: `GORILLA_ARTIFACT_GC_ENABLED=true`
-* Enable bucket versioning if you use [AWS](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html) or [GCP](https://cloud.google.com/storage/docs/object-versioning).
-* [Enable soft delete for blobs if you use Azure](https://learn.microsoft.com/en-us/azure/storage/blobs/soft-delete-blob-overview).
+Artifacts that are scheduled for deletion with a TTL policy, deleted with the W&B SDK, or deleted with the W&B App UI are first soft-deleted. Artifacts that are soft deleted undergo garbage collection before they are to marked for hard-deleted.
```
Comment (noaleetz) [inline docs/guides/artifacts/delete-artifacts.md]:
```suggestion
Artifacts that are scheduled for deletion with a TTL policy, deleted with the W&B SDK, or deleted with the W&B App UI are first soft-deleted. Artifacts that are soft deleted undergo garbage collection before they are hard-deleted.
```

**Review by abhinavg6 — APPROVED** [review]
LGTM, thanks for the changes.

**Review by noaleetz — APPROVED** [review]
ready to go!

File: docs/guides/hosting/intro.md [hunk 22]
```diff
@@ -22,6 +22,7 @@ With W&B Server you can configure and leverage features such as:
 - [User Management](./manage-users.md)
 - [Prometheus Monitoring](./prometheus-logging.md)
 - [Slack Alerts](./slack-alerts.md) and more.
```
Comment (szymon-piechowicz-wandb) [inline docs/guides/hosting/intro.md]: Move `and more.` to the last bullet point.

**Review by szymon-piechowicz-wandb — COMMENTED** [review]
We should add `GORILLA_ARTIFACT_GC_ENABLED` to https://docs.wandb.ai/guides/hosting/env-vars.

Btw, the documentation for `GORILLA_DATA_RETENTION_PERIOD ` there also needs to be improved. It says the data is deleted "immediately" but that's not true. That's exactly what the variable controls: the delay after a run is deleted, and before it gets garbage collected.
